### PR TITLE
Allow direct listening to host/port in geographic gatherer.

### PR DESCRIPTION
**Pull request by mraspaud**
This PR allows the user to provide a port and host for listening.

It also refactors the geographic gatherer a bit

### CODE REVIEW

File: pytroll_collectors/tests/test_geographic_gatherer.py [hunk 1]
```diff
@@ -43,271 +45,364 @@ def __init__(self, config_item=None, publish_port=0, nameservers=None):
...
+    def start(self):
+        """Start."""
+        self.start_called = True
+
+
+@pytest.fixture
+def create_publisher_from_dict_config():
+    """Return a faked function."""
+    with patch('pytroll_collectors.utils.create_publisher_from_dict_config') as patched_fun:
+        yield patched_fun
```
Comment (pnuu) [inline pytroll_collectors/tests/test_geographic_gatherer.py:76]: This fixture is not used.

File: pytroll_collectors/tests/test_geographic_gatherer.py [hunk 2]
```diff
@@ -43,271 +45,364 @@ def __init__(self, config_item=None, publish_port=0, nameservers=None):
...
+
 
-    def setUp(self):
+@pytest.fixture
+def ns_subscriber():
+    """Return a fake ns subscriber."""
+    with patch('pytroll_collectors.triggers._posttroll.NSSubscriber') as patched_fun:
+        yield patched_fun
```
Comment (pnuu) [inline pytroll_collectors/tests/test_geographic_gatherer.py:83]: This fixture is not used.

File: pytroll_collectors/tests/test_geographic_gatherer.py [hunk 3]
```diff
@@ -43,271 +45,364 @@ def __init__(self, config_item=None, publish_port=0, nameservers=None):
...
+
+
+@pytest.fixture
+def tmp_config_file(tmp_path):
+    """Return the path to the config file."""
+    return tmp_path / "config.ini"
+
+
+fake_create_publisher_from_dict_config = MagicMock()
+fake_sub_factory = MagicMock()
```
Comment (pnuu) [inline pytroll_collectors/tests/test_geographic_gatherer.py:93]: Isn't it a bit unsafe (race conditions) to use global variables as mocks?

**Review by pnuu — COMMENTED** [review]
Few comments inline.

Thanks again for the refactoring!

File: pytroll_collectors/tests/test_geographic_gatherer.py [hunk 4]
```diff
@@ -43,271 +45,364 @@ def __init__(self, config_item=None, publish_port=0, nameservers=None):
...
+        """Start."""
+        self.start_called = True
+
+
+class FakeWatchDogTrigger(WatchDogTrigger):
+    """Fake WatchDogTrigger."""
+
+    def publish_collection(self, *args, **kwargs):
+        """Publish collection."""
+        del args, kwargs
```
[the remark this answers is not in the document]
Comment (mraspaud) [reply pytroll_collectors/tests/test_geographic_gatherer.py:65]: fixed.

File: pytroll_collectors/tests/test_geographic_gatherer.py (hunk 1 above)
Comment (mraspaud) [reply pytroll_collectors/tests/test_geographic_gatherer.py:76]: fixed

File: pytroll_collectors/tests/test_geographic_gatherer.py (hunk 2 above)
Comment (mraspaud) [reply pytroll_collectors/tests/test_geographic_gatherer.py:83]: fixed

File: pytroll_collectors/tests/test_geographic_gatherer.py (hunk 3 above)
Comment (mraspaud) [reply pytroll_collectors/tests/test_geographic_gatherer.py:75]: pytest runs the tests sequentially, so I think it's fine

Comment (pnuu) [reply pytroll_collectors/tests/test_geographic_gatherer.py:75]: Oh, ok.

**Review by pnuu — APPROVED** [review]
LGTM

**Pull request merged**